pyproject.toml

```diff
@@ -6,5 +6,5 @@ requires = [
     "wheel",
     "cython",
     # Must be kept in sync with the `setup_requirements` in `setup.py`
-    'pyarrow>=3.0.0,<3.1.0',
+    'pyarrow>=4.0.1',
```

Comment:
This always needs an upper bound as the ABI breaks with every major release.

Reply:
From the discussion I see `This is blocking our scheduled release next week , we will upgrade OS version on our Mac build machine after release and commit this change again after that`

Does this mean I should wait until OS version is upgraded and after that we can use Arrow 4.x.x?